nixos/modules/config/users-groups.nix

```
@@ -592,6 +592,26 @@ in {
       '';
     };
 
+    # Warn about user accounts with deprecated password hashing schemes
+    system.activationScripts.hashes = {
+      deps = [ "users" ];
+      text = ''
+        users=()
+        while IFS=: read -r user hash tail; do
+          if [[ "$hash" != '!' && ! "$hash" =~ ^\$(y|gy|7|2b|2y|2a|6)\$ ]]; then
```

```suggestion
          if [[ "$hash" = '$'* && ! "$hash" =~ ^\$(y|gy|7|2b|2y|2a|6)\$ ]]; then
```
Arch uses `!*` for the field. Should skip unrecognized ones.

---

I cannot find format specification for `/etc/shadow`. Hopefully this is enough.